core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -1107,21 +1158,23 @@ public void testIncrementalCleanupFailsWhenExpiringSnapshotId() {
 
   @TestTemplate
   public void testMultipleRefsAndCleanExpiredFilesFailsForIncrementalCleanup() {
+    assumeThat(incrementalCleanup).isTrue();
+
     table.newAppend().appendFile(FILE_A).commit();
     table.newDelete().deleteFile(FILE_A).commit();
-    table.manageSnapshots().createTag("TagA", table.currentSnapshot().snapshotId()).commit();
-    waitUntilAfter(table.currentSnapshot().timestampMillis());
-    RemoveSnapshots removeSnapshots = (RemoveSnapshots) table.expireSnapshots();
```

Comment:
This had to change because now for this case we're being a bit more permissive. Previously any ref existiing would lead to a failure but this test was setting up a tag on main which got aged off. With the new logic we'd allow incremental cleanup here since the tag is on main and we can use incremental cleanup safely here. I updated the test so that it creates a branch and writes to the branch which is then aged off. The check that there are no snapshots being removed outside of main will fail in the new test.